contracts/Curation.sol

```diff
@@ -0,0 +1,415 @@
+pragma solidity 0.4.18;
+
+import "@aragon/os/contracts/apps/AragonApp.sol";
+import "@aragon/os/contracts/lib/misc/Migrations.sol";
+import "@aragon/os/contracts/lib/zeppelin/math/SafeMath.sol";
+import "@aragon/os/contracts/lib/zeppelin/math/SafeMath64.sol";
+
+import "@aragon/apps-staking/contracts/interfaces/IStaking.sol";
+import "./interfaces/IRegistry.sol";
+import "./interfaces/IVoting.sol";
+
+
+contract Curation is AragonApp {
+    using SafeMath for uint256;
+    using SafeMath64 for uint64;
+
+    uint64 constant public MAX_UINT64 = uint64(-1);
+    uint256 constant public PCT_BASE = 10 ** 18; // 0% = 0; 1% = 10^16; 100% = 10^18
+    bytes32 constant public MIN_DEPOSIT_PARAM = keccak256("MIN_DEPOSIT_PARAM");
+    bytes32 constant public APPLY_STAGE_LEN_PARAM = keccak256("APPLY_STAGE_LEN_PARAM");
+    bytes32 constant public DISPENSATION_PCT_PARAM = keccak256("DISPENSATION_PCT_PARAM");
+
+    IRegistry public registry;
+    IStaking public staking;
+    IVoting public voting;
+    uint256 public minDeposit;
+    uint64 public applyStageLen;
+    uint256 public dispensationPct;
+
+    bytes32 constant public CHANGE_PARAMS_ROLE = keccak256("CHANGE_PARAMS_ROLE");
+    bytes32 constant public CHANGE_VOTING_APP_ROLE = keccak256("CHANGE_VOTING_APP_ROLE");
+
+    // TODO!!!
+    enum TimeUnit { Blocks, Seconds }
+
+    struct Application {
+        address applicant;
+        uint64 date;
+        bool registered;
+        bytes data;
+        uint256 amount;
+        uint256 lockId;
+    }
+
+    struct Challenge {
+        address challenger;
+        uint64 date;
+        bool resolved;
+        uint256 amount;
+        uint256 lockId;
+        uint256 voteId;
+        uint256 dispensationPct;
+        mapping(address => bool) claims; // participants who already claimed their reward
+    }
+
+    mapping(bytes32 => Application) applications;
+    mapping(bytes32 => Challenge) challenges;
+    mapping(address => mapping(uint256 => bool)) usedLocks;
+
+    event NewApplication(bytes32 entryId, address applicant);
```

Comment:
We should index `entryId`s

Reply:
Btw, there already was an issue for this, #12 , I just forgot ;-) I'm going to add it to the description so it's closed automatically.